Project-Music-Data/README.md

```diff
@@ -0,0 +1,72 @@
+2. `getSong(songID)`: when called with one string as an argument, returns an an object containing information about a single song.
+
+## Requirements
+
+This is an individual project. You are expected to work on it on your own. You can talk to other trainees and volunteers and get help and advice, but you should write all of the code yourself.
+
+You must submit both a link to your GitHub repo, and a link to the deployed website.
+
+Your website must be hosted on the internet, and must be automatically deployed when you merge changes to your GitHub repo.
+
+Your website must include a drop-down to select a user to display information for. When a user is selected, you must display answers to all of the questions for that user. If a question doesn't apply to that user, you should not show the question.
+
+You **must not** implement any kind of authentication. Just a drop-down to choose which user's information to display.
+
+Your GitHub repository must contain unit tests which demonstrate that your code works. End to end tests are optional.
+
+## Questions to answer
+
+1. What was the user's most often listened to song according to the data?
+2. What was the user's most often listened to artist according to the data?
+3. What was the user's most often listened to song on Friday nights (between 5pm and 4am)?
+4. What are the answers to the above questions if using _listening time_ rather than _number of listens_?
+5. What song did the user listen to the most times in a row (i.e. without any other song being listened to in between)? How many times was it listened to?
+6. Are there any songs that the user listened to every day between the first day that user listened to any songs and the last day that user listened to any songs? If the answer is yes, you should show which one(s). If the answer is no, you should not show anything about this question.
+7. What were the user's top three genres to listen to?
+
+## Rubric
+
+All of the below requirements must be met for the project to be considered complete:
+
+* The website must contain a drop-down which lists three users.
+* Selecting a user must display answers relevant to that user (see table below).
+* The code written to calculate the answers to the questions must seem like it could handle different data if it were supplied, including the following edge-cases:
+  * An empty array of song data, or an empty subset for any question, should completely hide the question and answer. Displaying the question and an empty result, or any kind of error, is not acceptable. If no questions apply to the user, some intelligible statement should be shown to the user (e.g. "This user didn't listen to any songs.").
+  * If fewer than three (but more than zero) genres were listened to the site should list the top genres listened to, and should not display text like "Top 3 genres" but may say "Top genres" or "Top 2 genres" or similar.
+* Unit tests must be written for at least one non-trivial function.
```

Comment:
```suggestion
* Unit tests must be written for at least one non-trivial function.
* The HTML is valid
* There is no CSS
```

Reply:
I'm not sure about completely banning CSS. We've discussed before that extensions / stretch goals are fine, just we won't give people credit for them. This feels like it goes a step further, and actively fails people for doing a stretch goal, rather than just discouraging it?

Reply:
I think it's better to save them from themselves tbh.

Reply:
Perhaps a compromise position here is that we could mark a specific version as the submitted version - and allow for stretch goals past this point. IIRC there's wording about git shas in the project recommendations? Might also be a good opportunity to experience branching "for real"

Reply:
For now I've gone with the compromise of instructing them not to write CSS, but not adding to the rubric that they'll fail if they do.